cdk/lib/security-hq.ts

```diff
@@ -1,17 +1,170 @@
+import {
+  ComparisonOperator,
+  Metric,
+  TreatMissingData,
+} from '@aws-cdk/aws-cloudwatch';
+import type { CfnTable } from '@aws-cdk/aws-dynamodb';
+import { InstanceClass, InstanceSize, InstanceType } from '@aws-cdk/aws-ec2';
+import type { CfnTopic } from '@aws-cdk/aws-sns';
 import { CfnInclude } from '@aws-cdk/cloudformation-include';
+import { Duration } from '@aws-cdk/core';
 import type { App } from '@aws-cdk/core';
+import { AccessScope, GuApplicationPorts, GuEc2App } from '@guardian/cdk';
+import { GuAlarm } from '@guardian/cdk/lib/constructs/cloudwatch';
 import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
-import { GuStack, GuStageParameter } from '@guardian/cdk/lib/constructs/core';
+import {
+  GuDistributionBucketParameter,
+  GuStack,
+} from '@guardian/cdk/lib/constructs/core';
+import {
+  GuDynamoDBReadPolicy,
+  GuDynamoDBWritePolicy,
+} from '@guardian/cdk/lib/constructs/iam';
 
+/**
+ * Migration steps:
+ *
+ * - ensure new stack is working okay
+ * - make DNS switch (defined in domains account)
+ * - include full definitions for adopted resources (dynamo, alarm topic) + stop
+ *   importing old template
+ * - finally delete old template + stack
+ */
 export class SecurityHQ extends GuStack {
   migratedFromCloudFormation = true;
 
   constructor(scope: App, id: string, props: GuStackProps) {
     super(scope, id, props);
 
-    new CfnInclude(this, `security-hq-PROD`, {
+    // Import old stuff. Note, this template has been modified slightly to avoid
+    // a clash on the VpcId parameter - as the new service runs in a different
+    // VPC to the old.
+    const template = new CfnInclude(this, `security-hq-PROD`, {
       templateFile: '../cloudformation/security-hq.template.yaml',
-      parameters: { Stage: GuStageParameter.getInstance(this) },
+      parameters: {
+        Stage: this.stage,
+      },
+    });
+
+    // Import the existing template, though note we have removed some
+    // overlapping resources - such as the stage parameter.
+    const cfnTable = template.getResource(
+      'SecurityHqIamDynamoTable'
+    ) as CfnTable;
+
+    // TODO use below once old template is gone instead of importing above.
+    /*     const table = new Table(this, 'DynamoTable', {
+      tableName: `security-hq-iam-${this.stage}`,
+      removalPolicy: RemovalPolicy.RETAIN, // TODO set update+delete policy too.
+      readCapacity: 5,
+      writeCapacity: 5,
+      partitionKey: {
+        name: 'id',
+        type: AttributeType.STRING,
+      },
+    });
+    const defaultChild = table.node.defaultChild as unknown as CfnElement;
+    defaultChild.overrideLogicalId('SecurityHqIamDynamoTable'); */
+
+    // The new stack below...
+
+    const distBucket = GuDistributionBucketParameter.getInstance(this);
+
+    new GuEc2App(this, {
+      access: { scope: AccessScope.PUBLIC },
```

Comment:
Based on the [template](https://github.com/guardian/security-hq/blob/main/cloudformation/security-hq.template.yaml#L240-L245), I think this might be an unintentional change?

There are examples [here](https://guardian.github.io/cdk/classes/index.GuEc2App.html) which should help with this. AMIable also has an [example of this in action](https://github.com/guardian/amiable/blob/cae8784ed04022384fb0c85350ce7908d932be24/cdk/lib/amiable/amiable.ts#L37), which keeps the exact CIDR ranges being used private.

Reply:
Ah good spot, thanks.